app/src/main/java/com/javahelp/frontend/activity/ProviderRegistrationActivity.java

```diff
@@ -25,14 +50,109 @@ protected void onCreate(Bundle savedInstanceState) {
         binding.setPdata(providerRegistrationViewmodel);
         binding.setLifecycleOwner(this);
 
-        binding.signupbtn.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                Intent intent = new Intent(ProviderRegistrationActivity.this,
-                        FrontPage2Activity.class);
-                startActivity(intent);
-            }
-        });
+        providerRegistrationViewmodel.getRegisterResult().observe(this, this::updateOnRegisterResult);
+
+        binding.signupbtn.setOnClickListener(this::registerClick);
+
+    }
+
+    /**
+     * Called when the Register button is clicked
+     *
+     * @param v {@link View} that was clicked
+     */
+    private void registerClick(View v) {
+        providerRegistrationViewmodel.setUsername(binding.username.getText().toString());
+        providerRegistrationViewmodel.setPassword1(binding.password.getText().toString());
+        providerRegistrationViewmodel.setPassword2(binding.repassword.getText().toString());
+        providerRegistrationViewmodel.setProviderUserInfo(binding.email.getText().toString(),
+                binding.firstname.getText().toString(),
+                binding.lastname.getText().toString(),
+                binding.home.getText().toString(),
+                binding.phonenumber.getText().toString());
+        registerAttempt();
+    }
+
+    /**
+     * Initiates a register request
+     */
+    private void registerAttempt() {
+        if (binding.username.getText().toString().isEmpty() || binding.password.getText().toString().isEmpty()
+        || binding.repassword.getText().toString().isEmpty()) {
+            providerRegistrationViewmodel.setRegisterError("Please enter username and password");
+            return; // early return on invalid username or password
+        }else if(!binding.password.getText().toString().equals(binding.repassword.getText().toString())){
+            providerRegistrationViewmodel.setRegisterError("Passwords do not match");
+            return;
+            }else if(binding.email.getText().toString().isEmpty() || binding.firstname.getText().toString().isEmpty()
```

Comment:
This should be changed for `Providers` into `practiceName` and optional `isCertified`

Reply:
yes, but we do not have practicename for our registration UI, I just assume practicename is last name + first name.
Also, we do not have a certified option on the provider registration UI, and it is hard to actually implement the certified functionality, I think we just set it as false

Reply:
I guess yeah, we could get both `lastName` and `firstName` from the UI, then combine them both (`First` + `Last` name) together to make a `practiceName` in somewhere like `ProviderRegistrationViewmodel.java`? before sending it to the backend.

Reply:
We should add practice name to the registration UI for providers